Sources/MongoSwift/ConnectionString.swift

```diff
@@ -24,6 +24,38 @@ internal class ConnectionString {
             }
         }
 
+        if let compressors = options?.compressors {
+            // user specified an empty array, so we should nil out any compressors set via connection string.
+            guard !compressors.isEmpty else {
+                guard mongoc_uri_set_compressors(self._uri, nil) else {
```

Comment:
I'm not sure if it's actually OK/intended behavior that passing in nil here works. I did test passing in an empty string instead but that generates a libmongoc warning. I asked in #mongoc so we'll see what they say.

Reply:
update: Kevin said it's safe to rely on this, and he's opened [CDRIVER-3751](https://jira.mongodb.org/browse/CDRIVER-3751) about mentioning it in the documentation and adding a test to ensure it continues working.